### PR TITLE
Extend process_notebooks for testing

### PR DESCRIPTION
**Pull request by jackgerrits**
## Why are these changes needed?

- Enables process_notebooks to test notebooks too
- Also removes the manual quarto render step

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: website/process_notebooks.py [hunk 1]
```diff
@@ -121,56 +164,166 @@ def skip_reason_or_none_if_ok(notebook: Path) -> typing.Optional[str]:
...
+
+@dataclass
+class NotebookSkip:
+    reason: str
+
+
+NB_VERSION = 4
+
+
+def test_notebook(notebook_path: Path, timeout: int = 300) -> Tuple[Path, Optional[Union[NotebookError, NotebookSkip]]]:
```
Comment (sonichi) [inline website/process_notebooks.py:256]: Could you document in notebook contribution guide about these options? Also add a link in Contribute.md.

Comment (jackgerrits) [reply website/process_notebooks.py:256]: Updated

**Review by ekzhu — APPROVED** [review]

File: notebook/contributing.md [hunk 2]
```diff
@@ -74,3 +74,35 @@ Learn more about configuring LLMs for agents [here](/docs/llm_configuration).
...
+python website/process_notebooks.py test notebook/agentchat_logging.ipynb
+```
+
+Options:
+- `--timeout` - timeout for a single notebook
+- `--exit-on-first-fail` - stop executing further notebooks after the first one fails
+
+### Skip tests
+
+If a notebook needs to be skipped then add to the notebook metadata:
```
Comment (ekzhu) [inline notebook/contributing.md:100]: Nit: some people may not know how to edit the metadata. Could add "You can open the notebook file in a text editor or "Text Editor" mode in VS Code.

File: website/build_website.sh [hunk 3]
```diff
@@ -28,11 +28,8 @@ fi
 # Generate documentation using pydoc-markdown
 pydoc-markdown
 
-# Render the website using Quarto
-quarto render ./docs
-
 # Process notebooks using a Python script
-python ./process_notebooks.py
+python ./process_notebooks.py render
```
Comment (ekzhu) [inline website/build_website.sh:32]: Nit: no shebang here? 😺

File: notebook/contributing.md (hunk 2 above)
Comment (jackgerrits) [reply notebook/contributing.md:100]: Good point - this is about to get more important as the front matter is going to be moved there in my next PR

File: website/build_website.sh (hunk 3 above)
Comment (jackgerrits) [reply website/build_website.sh:32]: Hmm yeah we could do that

**Pull request merged**